src/index.js

```diff
@@ -20,70 +22,95 @@ module.exports.args = {
 }
 
 function plugin (browserify, pluginOpts) {
+  //pluginOpts.config is config path
+  const configuration = {
+    writeAutoConfig: pluginOpts.writeAutoConfig,
+    getConfig: () => {
+      return getConfigFrom(pluginOpts)
+    },
+    configOverride: pluginOpts.configOverride,
+    generateAutoConfig: pluginOpts.autoConfig,
+    autoConfigPath: getConfigPath(pluginOpts)
+  }
   // setup the plugin in a re-bundle friendly way
   browserify.on('reset', setupPlugin)
   setupPlugin()
-
   // override browserify/browser-pack prelude
   function setupPlugin () {
-
+    
     applySesTransforms(browserify)
-
-    // helper to read config at path
-    if (typeof pluginOpts.config === 'string') {
-      pluginOpts.lavamoatConfig = () => {
-        if (pluginOpts.writeAutoConfig) {
-          return
-        }
-        // load latest config
-        const filename = pluginOpts.config
-        const configSource = fs.readFileSync(filename, 'utf8')
-        // if override specified, merge
-        if (pluginOpts.configOverride) {
-          const filename = pluginOpts.configOverride
-          const configOverrideSource = fs.readFileSync(filename, 'utf8')
-          const initialConfig = JSON.parse(configSource)
-          const overrideConfig = JSON.parse(configOverrideSource)
-          const config = mergeDeep(initialConfig, overrideConfig)
-          return config
-        }
-        return configSource
-      }
-    } else if (pluginOpts.config !== undefined) {
-      // "config" for alias "lavamoatConfig"
-      pluginOpts.lavamoatConfig = pluginOpts.config
-    }
-
-    const customPack = createLavamoatPacker(pluginOpts)
+    
+    const customPack = createLavamoatPacker(configuration)
     // replace the standard browser-pack with our custom packer
     browserify.pipeline.splice('pack', 1, customPack)
-
+    
     // inject package name into module data
     browserify.pipeline.splice('emit-deps', 0, createPackageDataStream())
-
     // helper to dump autoconfig to a file
-    if (pluginOpts.writeAutoConfig) {
-      const filename = pluginOpts.config
-      if (typeof filename !== 'string') {
+    if (configuration.writeAutoConfig) {
+      if (typeof configuration.autoConfigPath !== 'string') {
         throw new Error('LavaMoat - "writeAutoConfig" was specified but "config" is not a string')
       }
-      pluginOpts.autoConfig = function writeAutoConfig (config) {
-        fs.writeFileSync(filename, config)
-        console.warn(`LavaMoat Autoconfig - wrote to "${filename}"`)
+
+      if (!fs.existsSync(configuration.autoConfigPath)) {
+        const configDirectory = path.dirname(configuration.autoConfigPath)
+        mkdirp.sync(configDirectory)
+      }
+
+      configuration.generateAutoConfig = function writeAutoConfig (autoConfig) {
```

Comment:
do not modify the configuration object during plugin runtime
this should be moved above runPlugin